### ISSUE TITLE
Action bar has broken accessibility

### ISSUE THREAD

**Issue by saharso** [microsoft/vscode#100273]
Issue Type: <b>Bug</b>

I often use tab navigation in the git panel. Usually I navigate to the commit button (the one with the V icon) with shift+tab after writting the commit message, and then press enter to send the commit. Recently, this stopped working, and I am forced to use the mouse in order to submit a commit.

VS Code version: Code 1.46.0 (a5d1cc28bb5da32ec67e86cc50f84c67cc690321, 2020-06-10T08:59:06.977Z)
OS version: Darwin x64 18.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|8.00GB (0.04GB free)|
|Process Argv|-psn_0_19292773|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
spellright|ban|3.0.50
vscode-eslint|dba|2.1.5
githistory|don|0.6.5
gitlens|eam|10.2.2
beautify|Hoo|1.5.0
vscode-typescript-tslint-plugin|ms-|1.2.3
vscode-jest|Ort|3.2.0
code-spell-checker|str|1.9.0
vscodeintellicode|Vis|1.2.8

</details>

**Comment by gjsjohnmurray** [comment]
It seems that when focus is on one of those buttons, pressing Enter triggers the action of the button immediately to the right of the focused one. For example, focus here:

![image](https://user-images.githubusercontent.com/6726799/84764529-c5319300-afc5-11ea-9823-c99f21db70e2.png)

Then press Enter. The '...' menu appears.

Focus here and press Enter:
![image](https://user-images.githubusercontent.com/6726799/84764621-ed20f680-afc5-11ea-8bb3-786fbf7cbadf.png)

A refresh occurs.

Focus here (Toggle View Mode) and press Enter:
![image](https://user-images.githubusercontent.com/6726799/84764743-1b9ed180-afc6-11ea-824f-efb2898a6203.png)

This attempts a commit (i.e. as though you clicked the checkmark button).

I don't get the problem on 1.47.0-insider. On that version the Toggle View Mode button has been removed. Instead there's a 'View & Sort' submenu from '...'

![image](https://user-images.githubusercontent.com/6726799/84765255-e0e96900-afc6-11ea-9fbd-e96c5b078628.png)

So maybe the problem was resolved as part of that work.

**Comment by joaomoreno** [comment]
@sbatten @isidorn Does this ring any of your bells?

**Comment by joaomoreno** [comment]
I can't reproduce this in either latest stable or latest insiders.

@gjsjohnmurray Which version did you reproduce this on?

**Comment by joaomoreno** [comment]
Did any of you press any arrow keys? I can repro somewhat that pressing left/right arrow will actually change the action without updating the UI.

**Comment by joaomoreno** [comment]
These steps show an issue which definitely sounds related:

1. focus scm input box
2. <kbd>Shift Tab</kbd>, to focus `...`
3. <kbd>LeftArrow</kbd>, nothing happens
4. <kbd>Enter</kbd>, git refresh is executed instead of `...`

**Comment by isidorn** [comment]
Just copy pasting our chat here for clarity

I can not repro with insiders
I can repro with stable
Now trying with stable recovery cnadidate since it has my fix
I can repro with stable recovery
So my fix for https://github.com/microsoft/vscode/issues/99907 did not fix this 
Some other thing fixed it

**Comment by gjsjohnmurray** [comment]
@joaomoreno here's an animated GIF

![junk](https://user-images.githubusercontent.com/6726799/84782057-7b55a680-afdf-11ea-968b-690c3e33be0c.gif)

Version: 1.46.0 (user setup)
Commit: a5d1cc28bb5da32ec67e86cc50f84c67cc690321
Date: 2020-06-10T09:03:20.462Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

**Comment by roblourens** [comment]
@isidorn can you check whehter 4d0976ca834d02e415d009093671d94d75f818eb fixed it?

**Comment by isidorn** [comment]
@roblourens when I revert that commit the issue is still fixed out of source. So that commit did not fix it.
If we want to fix this for the recovery somebody should just do some git bisection.
If we do not want to fix for the recovery than we can just close it since it is fixed in latest.

**Comment by roblourens** [comment]
Are you using these steps with the arrow keys to repro? Or just tabbing, or anything else?

**Comment by joaomoreno** [comment]
OK, this is a LONG one:

- The commit c630abdfcf96157dacb03f35339255b10be43559 is what made the issue _disappear_ from `master` but had nothing to do with the actual fix
- The bug originally comes from d9f1e1d684308446437a267dd2a3fa3fe66a2d94 in which @jrieken changed the semantics of calling `dispose(array)`, which used to return an empty array and now simply returns the same array still filled with the disposed elements
- @roblourens somehow figured that out and did a local fix with https://github.com/microsoft/vscode/commit/4d0976ca834d02e415d009093671d94d75f818eb and https://github.com/microsoft/vscode/commit/2d53adcb5289b10a40140f6f0084a8a7ac72e119

@jrieken I would reevaluate that decision to not return an empty array from `dispose`, this is definitely not the only place where this behavior was expected, I know I depend on it across my areas.

I suggest this to be fixed for the release, since it breaks accessibility and a handful of users already filed issues. As for the fix we have two possibilities:

1. Bring back the `dispose(array) === []` semantics, or
2. Bring @roblourens's commits to the release branch

I'll leave the decision to both of you.

**Comment by jrieken** [comment]
> Bring back the dispose(array) === [] semantics, or

That sounds like the safest options. The change is minimal and we don't know what else could be affected by this. Not returning an empty array wasn't intended.

**Comment by jrieken** [comment]
Merged PR and also cherry-picked bb55416b411deb87170c4460ebfbc369483006fb for master

**Comment by roblourens** [comment]
I forgot that dispose worked that way when I was looking at this, I should have checked git blame for dispose too. I'll revert my commits to clean up.
